insights/parsers/jboss_standalone_main_conf.py

```diff
@@ -0,0 +1,109 @@
+                    </authorization>
+                </security-realm>
+            </security-realms>
+        </management>
+    </server>
+
+Examples:
+
+    >>> from insights.tests import context_wrap
+    >>> from insights.parsers.jboss_standalone_main_conf import JbossStandaloneConf
+    >>> JBOSS_STANDALONE_CONFIG = '''
+    ... <server xmlns="urn:jboss:domain:1.7">
+    ...     <management>
+    ...         <security-realms>
+    ...             <security-realm name="ManagementRealm">
+    ...                 <authentication>
+    ...                     <local default-user="$local" skip-group-loading="true"/>
+    ...                     <properties path="mgmt-users.properties" relative-to="jboss.server.config.dir"/>
+    ...                 </authentication>
+    ...                 <authorization map-groups-to-roles="false">
+    ...                     <properties path="mgmt-groups.properties" relative-to="jboss.server.config.dir"/>
+    ...                 </authorization>
+    ...             </security-realm>
+    ...             <security-realm name="ApplicationRealm">
+    ...                 <authentication>
+    ...                     <local default-user="$local" allowed-users="*" skip-group-loading="true"/>
+    ...                     <properties path="application-users.properties" relative-to="jboss.server.config.dir"/>
+    ...                 </authentication>
+    ...                 <authorization>
+    ...                     <properties path="application-roles.properties" relative-to="jboss.server.config.dir"/>
+    ...                 </authorization>
+    ...             </security-realm>
+    ...         </security-realms>
+    ...     </management>
+    ... </server>
+    ... '''.strip()
+    >>> jboss_main_config = JbossStandaloneConf(context_wrap(JBOSS_STANDALONE_CONFIG, path="/root/jboss/jboss-eap-6.4/standalone/configuration/standalone.xml"))
+    >>> jboss_main_config.file_path
+    '/root/jboss/jboss-eap-6.4/standalone/configuration/standalone.xml'
+    >>> properties = sorted(jboss_standalone_conf.get_properties("management/security-realms/security-realm/authentication/properties"))
+    >>> properties[0].get("path")
+    'application-users.properties'
+    >>> properties[0].get("relative-to")
+    'jboss.server.config.dir'
+
+"""
+
+from .. import Parser, parser
+
+import xml.etree.ElementTree as ET
+from insights.specs import jboss_standalone_main_config
+
+
+@parser(jboss_standalone_main_config)
+class JbossStandaloneConf(Parser):
+    """
+    Wraps a DOM of standalone.xml
+
+    self.dom is an instance of ElementTree.
+    self.xmlns is the using namespace
+    """
+
+    def parse_content(self, content):
+        dom = ET.fromstring("\n".join(content))
+        self.dom = dom
+        xmlns = dom.tag.split("server")[0]
+        self.xmlns = xmlns
+
+    def get_properties(self, property_path):
```

Comment:
Doesn't the DOM support XPath searching?  Maybe we should get users to fetch properties using XPath queries?

Reply:
@PaulWay A good suggestion. According to https://docs.python.org/2/library/xml.etree.elementtree.html,  limited support XPath syntax is supported.  Xiangce and I agreed to create a base class for all XML related parsers in . So I will enhance this func when https://github.com/RedHatInsights/insights-core/pull/855 is ready to merge.